lib/Basic/Subprocess.cpp

```diff
@@ -1056,11 +1056,20 @@ void llbuild::basic::spawnProcess(
     assert(readfds[0].fd == outputPipeParentEnd.unsafeDescriptor());
     assert(readfds[1].fd == controlPipeParentEnd.unsafeDescriptor());
 
-    if (poll(readfds, nfds, -1) == -1) {
-      int err = errno;
-      delegate.processHadError(ctx, handle,
-          Twine("failed to poll (") + strerror(err) + ")");
-      break;
+    while (true) {
+      if (poll(readfds, nfds, -1) == -1) {
```

Comment:
You can move this `if` condition into the `while` condition, and avoid writing the `else break` and `while true`.